specification/sqlvirtualmachine/SqlVirtualMachine.Management/SqlVirtualMachine.tsp

```diff
@@ -0,0 +1,148 @@
+/**
+ * A SQL virtual machine.
+ */
+@Azure.ResourceManager.Private.armResourceInternal(SqlVirtualMachineProperties)
+@Http.Private.includeInapplicableMetadataInPayload(false)
+model SqlVirtualMachine extends Foundations.TrackedResource {
+  ...ResourceNameParameter<
+    Resource = SqlVirtualMachine,
+    KeyName = "sqlVirtualMachineName",
+    SegmentName = "sqlVirtualMachines",
+    NamePattern = "^((?!_)[^\\\\/\"'\\[\\]:|<>+=;,?*@&]{1,64}(?<![.-]))$"
+  >;
+
+  /**
+   * DO NOT USE. This value will be deprecated. Azure Active Directory identity of the server.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  identity?: ResourceIdentity;
+
+  @doc("The resource-specific properties for this resource.")
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  @Azure.ResourceManager.Private.armResourcePropertiesOptionality(true)
+  properties?: SqlVirtualMachineProperties;
+}
+
+@armResourceOperations
+interface SqlVirtualMachines {
+  /**
+   * Gets a SQL virtual machine.
+   */
+  get is ArmResourceRead<
+    SqlVirtualMachine,
+    Parameters = {
+      /**
+       * The child resources to include in the response.
+       */
+      @query("$expand")
+      $expand?: string;
+    }
+  >;
+
+  /**
+   * Creates or updates a SQL virtual machine.
+   */
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<SqlVirtualMachine>;
+
+  /**
+   * Updates SQL virtual machine tags.
+   */
+  @patch(#{ implicitOptionality: false })
+  update is ArmCustomPatchAsync<
+    SqlVirtualMachine,
+    PatchModel = SqlVirtualMachineUpdate,
+    Response = ArmResponse<SqlVirtualMachine> &
+      ArmAsyncOperationHeader<FinalResult = SqlVirtualMachine>
+  >;
+
+  /**
+   * Deletes a SQL virtual machine.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "For backward compatibility"
+  delete is ArmResourceDeleteWithoutOkAsync<
+    SqlVirtualMachine,
+    Response = ArmDeletedResponse | ArmDeleteAcceptedLroResponse | ArmDeletedNoContentResponse
+  >;
+
+  /**
+   * Gets all SQL virtual machines in a resource group.
+   */
+  listByResourceGroup is ArmResourceListByParent<SqlVirtualMachine>;
+
+  /**
+   * Gets all SQL virtual machines in a subscription.
+   */
+  list is ArmListBySubscription<SqlVirtualMachine>;
+
+  /**
+   * Starts SQL best practices Assessment on SQL virtual machine.
+   */
+  startAssessment is ArmResourceActionAsyncBase<
+    SqlVirtualMachine,
+    void,
+    ArmAcceptedLroResponse,
+    BaseParameters = Azure.ResourceManager.Foundations.DefaultBaseParameters<SqlVirtualMachine>
+  >;
+
+  /**
+   * Starts SQL best practices Assessment with Disk Config rules on SQL virtual machine
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("SqlVirtualMachines_FetchDCAssessment")
```

Comment:
why do you add `@operationId` on this op? It is already in `SqlVirtualMachines` interface and have the right name.

Reply:
Because it is upper case "F" in "FetchDCAssessment"